instaclustr/resource_vpc_peering.go

```diff
@@ -77,32 +75,14 @@ func resourceVpcPeering() *schema.Resource {
 }
 
 func resourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
-	log.Printf("[INFO] Creating VPC peering request.")
 	client := meta.(*Config).Client
 
-	const ClusterReadInterval = 5
-	const WaitForClusterTimeout = 60
-	var cdcID string
-	var latestStatus string
-	timePassed := 0
-	for {
-		cluster, err := client.ReadCluster(d.Get("cluster_id").(string))
-		if err != nil {
-			return fmt.Errorf("[Error] Error retrieving cluster info: %s", err)
-		}
-		latestStatus = cluster.ClusterStatus
-		if cluster.DataCentres[0].CdcStatus == "PROVISIONED" || cluster.ClusterStatus == "RUNNING" {
-			cdcID = cluster.DataCentres[0].ID
-			break
-		}
-		if timePassed > WaitForClusterTimeout {
-			return fmt.Errorf("[Error] Timed out waiting for cluster to have the status 'PROVISIONED' or 'RUNNING'. Current cluster status is '%s'", latestStatus)
-		}
-		time.Sleep(ClusterReadInterval * time.Second)
-		timePassed += ClusterReadInterval
+	cdcID, err := VpcPeeringCreate(d, meta)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating VPC peering request: %s", err)
```

Comment:
**[Question]** This might just be me lacking understanding, but this error message doesn't seem to line up with the func call above, this seems like it failed to create the VPC itself?

Reply:
The `VpcPeeringCreate ` func has common elements that are used by both GCP and AWS for creating a peering request and that func return cdcID which is when null gives this failed to create VPC error message

Reply:
So this function creates a request object? Not an actual request? That doesn't line up with what the function is by it's name.

Reply:
I think you should reword these error messages really, having identical error messages in 3 different places makes debugging pretty difficult

Reply:
Just gonna change the error message to `[Error] Error creating VPC peering request object `